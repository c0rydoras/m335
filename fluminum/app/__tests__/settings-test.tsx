import { render } from "@testing-library/react-native";
import { ToggleGroup } from "~/components/ui/toggle-group";
import { ToggleGroupItem } from "~/components/ui/toggle-group";
import { Text } from "~/components/ui/text";

test("Toggle group", () => {
    const tree = render(
        <ToggleGroup onValueChange={() => {}} value='deg' type="single">
            <ToggleGroupItem value="deg">
                <Text>
                    Deg
                </Text>
            </ToggleGroupItem>
            <ToggleGroupItem value="rad">
                <Text>
                    Rad
                </Text>
            </ToggleGroupItem>
            <ToggleGroupItem value="percent">
                <Text>
                    Percent
                </Text>
            </ToggleGroupItem>
        </ToggleGroup>
    );
    expect(tree).toMatchSnapshot();
  });