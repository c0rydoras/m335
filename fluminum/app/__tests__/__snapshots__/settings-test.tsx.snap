// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle group 1`] = `
<View
  className="flex flex-row items-center justify-center gap-1"
  role="group"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": true,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="web:group web:inline-flex items-center justify-center rounded-md web:ring-offset-background web:transition-colors web:hover:bg-muted active:bg-muted web:focus-visible:outline-none web:focus-visible:ring-2 web:focus-visible:ring-ring web:focus-visible:ring-offset-2 h-10 px-3 native:h-12 native:px-[12] bg-accent"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="radio"
  >
    <Text
      className="web:select-text text-sm native:text-base font-medium text-accent-foreground"
    >
      Deg
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": false,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="web:group web:inline-flex items-center justify-center rounded-md web:ring-offset-background web:transition-colors web:hover:bg-muted active:bg-muted web:focus-visible:outline-none web:focus-visible:ring-2 web:focus-visible:ring-ring web:focus-visible:ring-offset-2 bg-transparent h-10 px-3 native:h-12 native:px-[12]"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="radio"
  >
    <Text
      className="web:select-text text-sm native:text-base text-foreground font-medium web:group-hover:text-muted-foreground"
    >
      Rad
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": false,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="web:group web:inline-flex items-center justify-center rounded-md web:ring-offset-background web:transition-colors web:hover:bg-muted active:bg-muted web:focus-visible:outline-none web:focus-visible:ring-2 web:focus-visible:ring-ring web:focus-visible:ring-offset-2 bg-transparent h-10 px-3 native:h-12 native:px-[12]"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="radio"
  >
    <Text
      className="web:select-text text-sm native:text-base text-foreground font-medium web:group-hover:text-muted-foreground"
    >
      Percent
    </Text>
  </View>
</View>
`;
